README.md

```diff
@@ -145,7 +145,7 @@ fastify.post('/', async function (req, reply) {
 fastify.post('/upload/raw/any', async function (req, reply) {
   const parts = req.parts()
   for await (const part of parts) {
-    if (part.file) {
+    if ('file' in part) {
```

Comment:
It is in JavaScript, but this way typescript narrows the type of "part" from "Multipart" to "MultipartFile"

Reply:
`'file' in part` check for existence, not is the value is truthy.

```js
'file' in { file: undefined } // true

({ file: undefined }).file // false
```

Reply:
This change is not appropriate to fix your issue. See the example that @climba03003 gave.

Reply:
If this claim is true, you could add a typescript test and add in the if clause: 

expectType<MultipartFile>(part)

Reply:
From what we mean is that, if the check is needed it should be `type` assertion.
For example, `if(part.file && typeof part.file === 'object')`. It is exist and something we want.

`'file' in part` do not guarantee it is something you want.

Reply:
In typescript world I would actually do:

```typescript
function isMultipartFile(part: unknown): part as MultipartFile {
  return (
    part.file &&
    part.file === 'object'
  )
}
```

Reply:
TypeScript need [type discriminator](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#the-in-operator-narrowing) for `MultipartFile` type and `Multipart` type. like that,

```ts
interface MultipartFile {
  kind: 'multipart-file'
  ...
}

interface Multipart {
  kind: 'multipart'
}
```

but in this case, MultipartFile and Multipart don't have this discriminator. So I feel resonable from ` if ('file' in part && part.file) {}`. But documentation changing different problem. Because JavaScript don't need type discriminator and type narrowing. Perhaps it would be nice to add 'TypeScript' to another section like that: [Fastify TypeScript](https://www.fastify.io/docs/latest/Reference/TypeScript/)

Reply:
Typescript does not need strictly type discriminators. Thats why you can add type guards. Also why should this belong in the general fastify typescript documentation?

Reply:
Multipart can be `MultipartFile` and `MultipartValue`. `MultipartValue` doesn't have `file` field. Maybe,

```ts
  for await (const part of parts) {
    await pump(part.file, fs.createWriteStream(part.filename))
  }
``` 

Maybe this code will be error in TypeScript. Because TypeScript compiler cannot determine `part` is `MultipartFile` and `MultipartValue`. So need strictly type discriminators (in case of compiler option `strict` set `true`)

Reply:
> From what we mean is that, if the check is needed it should be `type` assertion. For example, `if(part.file && typeof part.file === 'object')`. It is exist and something we want.
> 
> `'file' in part` do not guarantee it is something you want.

`if(part.file && typeof part.file === 'object')` gives the same typescript error `Property 'file' does not exist on type 'Multipart'.`